lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -689,23 +689,38 @@ def install_puppet_from_dmg_on( hosts, opts )
         # @return nil
         # @api private
         def install_puppet_agent_from_dmg_on(hosts, opts)
+          require 'nokogiri'
```

Comment:
@kevpl I was attempting to preserve the internal expectation of support for `latest` as an introspected version as stated by the documentation for this function.

"latest" is not a valid version for published puppet-agent dmg files.

I would also prefer to remove the dependence on Nokogiri. If there is a way to achieve this without changing the API of this method, I am open to suggestions. I am not sure how to get that data without scraping the HTML for the `opts[:mac_download_url]`.